metadata/src/main/resources/common/metadata/PartitionChangeRecord.json

```diff
@@ -40,6 +42,15 @@
       "versions": "0+", "nullableVersions": "0+", "taggedVersions": "0+", "tag": 4,
       "about": "null if the adding replicas didn't change; the new adding replicas otherwise." },
     { "name": "LeaderRecoveryState", "type": "int8", "default": "-1", "versions": "0+", "taggedVersions": "0+", "tag": 5,
-      "about": "-1 if it didn't change; 0 if the leader was elected from the ISR or recovered from an unclean election; 1 if the leader that was elected using unclean leader election and it is still recovering." }
+      "about": "-1 if it didn't change; 0 if the leader was elected from the ISR or recovered from an unclean election; 1 if the leader that was elected using unclean leader election and it is still recovering." },
+    { "name": "EligibleLeaderReplicas", "type": "[]int32", "default": "null", "entityType": "brokerId",
+      "versions": "1+", "nullableVersions": "1+", "taggedVersions": "1+", "tag": 6,
+      "about": "null if the ELR didn't change; the new eligible leader replicas otherwise." },
+    { "name": "LastKnownELR", "type": "[]int32", "default": "null", "entityType": "brokerId",
+      "versions": "1+", "nullableVersions": "1+", "taggedVersions": "1+", "tag": 7,
+      "about": "null if the LastKnownELR didn't change; the last known eligible leader replicas otherwise." },
+    { "name": "LastKnownLeader", "type": "int32", "default": "-2", "entityType": "brokerId",
```

Comment:
I see we want to convey one of three things:

* An updated LastKnownLeader (some value 0+)
* An empty LastKnownLeader (-1)
* An unchanged LastKnownLeader

The convention we have established in this record is to use the absence of a tagged field as "unchanged". Since these fields are only available at record version 1+, can we make the "unchanged" condition: "record version is 1+ and LastKnownLeader is absent" -- would that work?

Reply:
Synced offline, make the lastKnownLeader a tagged field.